### PR TITLE
Refactoring with Dapper

### PR DESCRIPTION
**Pull request by georgiev84**

### CODE REVIEW

File: Warehouse.Persistence.Abstractions/Constants/DapperConstants.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Warehouse.Persistence.Abstractions.Constants;
+
+public class DapperConstants
+{
+    public const string GetById = """SELECT * FROM "{0}" WHERE "Id" = @Id""";
```
Comment (Admiraculix) [inline Warehouse.Persistence.Abstractions/Constants/DapperConstants.cs]: `$"""SELECT * FROM "{entity}"  WHERE "Id" = @id """;`

image

File: Warehouse.Application/Features/Commands/Products/ProductDelete/ProductDeleteCommandHandler.cs [hunk 2]
```diff
@@ -8,14 +8,25 @@ public class ProductDeleteCommandHandler(IUnitOfWork _unitOfWork) : IRequestHand
...
-        await _unitOfWork.SaveAsync();
+            _unitOfWork.Products.Delete(existingProduct);
+            _unitOfWork.Commit();
+        }
+        catch (Exception ex)
+        {
+            _unitOfWork.Rollback();
+            throw new ApplicationException("An error occurred while processing the request.", ex);
+        }
+        finally
```
Comment (Admiraculix) [inline Warehouse.Application/Features/Commands/Products/ProductDelete/ProductDeleteCommandHandler.cs]: wouldn't this construction be better:
try/catch/finally be part of the unit of work's commit method so we don't need to put it in every handler.
What do you think of as an approach?

File: Warehouse.Persistence.Abstractions/Constants/DapperGenericConstants.cs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Warehouse.Persistence.Abstractions.Constants;
+
+public class DapperGenericConstants
+{
+    public const string GetById = """SELECT * FROM "{0}" WHERE "Id" = @Id""";
```
Comment (Admiraculix) [inline Warehouse.Persistence.Abstractions/Constants/DapperGenericConstants.cs]: `public static string GetById(string entity ) => """SELECT * FROM "{entity}" WHERE "Id" = @Id""";`
may be something like that

File: Warehouse.Persistence.PostgreSQL/Configuration/Contstants/DapperBasketConstants/DapperBasketReadConst.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+﻿namespace Warehouse.Persistence.PostgreSQL.Configuration.Contstants.DapperBasketConstants;
+public static class DapperBasketReadConst
```
Comment (Admiraculix) [inline Warehouse.Persistence.PostgreSQL/Configuration/Contstants/DapperBasketConstants/DapperBasketReadConst.cs]: I want to divide them by Mutated state and read state to follow the approach of CQRS:
MutatableQueryContant.cs  -> Create, Update, Delete -> MutatableQueryContant.MutateProductConstant
ReabableQueryConstant.cs  -> Read  -> > ReadableQueryContant.ReadProductConstant
or folder 
Constants/MutatableQueries/MutatableQueryProductConstant.cs
Constants/ReadableQueries/ReadbleQueryProductConstant.cs

Or something like that

**Review by Admiraculix — CHANGES_REQUESTED** [review]

File: Warehouse.Persistence.Abstractions/Constants/DapperGenericConstants.cs (hunk 3 above)
Comment (georgiev84) [reply Warehouse.Persistence.Abstractions/Constants/DapperGenericConstants.cs]: I prefer to use current solution to be consistent with all other queries. I will just arrange the query to different rows.

File: Warehouse.Persistence.PostgreSQL/Configuration/Contstants/DapperBasketConstants/DapperBasketReadConst.cs (hunk 4 above)
Comment (georgiev84) [reply Warehouse.Persistence.PostgreSQL/Configuration/Contstants/DapperBasketConstants/DapperBasketReadConst.cs]: image
I prefer this one

**Review by Admiraculix — APPROVED** [review]

**Review by Admiraculix — APPROVED** [review]

**Review by Admiraculix — APPROVED** [review]

**Pull request merged**